### DISCUSSION TITLE
How to perform search by hastags?

### DISCUSSION THREAD

**Discussion by liquidcarbon** [MarshalX/atproto#423] [Questions] [answered]
Hello, I'm trying to understand how search works, following https://github.com/MarshalX/atproto/blob/main/packages/atproto_client/models/app/bsky/feed/search_posts.py

```
r = client.app.bsky.feed.search_posts(
    params=dict(
        q="py",
        # tag=["python"],  # <--- this does not do anything
        lang="en",
        sort="top",
        limit=5
    )
)
```

Would you expect this to work? If not, what's the way to get roughly the results you would find here? https://bsky.app/hashtag/python

**Comment by MarshalX** [comment]
Hello, the link that you provided uses `app.bsky.feed.searchPosts?q=#python&limit=25&sort=top`

<img width="1568" alt="image" src="https://github.com/user-attachments/assets/7a2ff67d-f54c-4028-8943-f8b5e41baab8">

**Comment by liquidcarbon** [comment]
Thanks for the dev tools trick!

I tried other documented Params such as lang, since, until, and they work as expected.  Was tags documented but not implemented?

**Comment by MarshalX** [comment, marked as answer]
My pleasure! Well, it is implemented in SDK for sure, but looks like not implemented on server side. I am not related to server side part. You can try to dug into server side code of search in case you need the answer :) Backend code is here https://github.com/bluesky-social/atproto
